File: totosuki/ABC/B/084.py
A, B = map(int, input().split())
S = input()
rslt = "Yes"

for i in range(A):
  if S[i] in "0123456789":
    continue
  else:
    rslt = "No"
if S[A] != "-":
  rslt = "No"
for i in range(B):
  if S[-1 * i] in "0123456789":
    continue
  else:
    rslt = "No"

print(rslt)

File: OCH_Tao/ABC/B/084.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const [A, B] = (input[0].split(" ")).map(x => parseInt(x, 10));
const S = [...(input[1])];
const X = S.splice(A, 1);
if (X == "-") {
  if (!isNaN(S.join(""))) {
    console.log("Yes");
  } else {
    console.log("No");
  }
} else {
  console.log("No");
}